### PR TITLE
Add SARIF parser

### PR DESCRIPTION
**Pull request by tushar-deepsource**
- Adds independent `sarif-parser` packaage, with tests and a CLI.
- Adds a CLI in `run_community_analyzer.py` which passes the correct `issue_map.json`, and a test for artifacts.
- Adds tox config and CI with reporting test coverage to DeepSource.

### CODE REVIEW

File: sarif-parser/tests/cli_test.py [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+        "location": {
+            "path": "workspace/git/myproject/asd.yaml",
+            "position": {
+                "begin": {"line": 1, "column": 1},
+                "end": {"line": 1, "column": 1},
+            },
+        },
+    },
+    {
+        "issue_code": "container-resources",
```
Comment (srijan-deepsource) [inline sarif-parser/tests/cli_test.py:114]: Is this made up, or are we missing this in the mapping?

File: tests/community_analyzer_test.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    run_community_analyzer.main(["--analyzer=kube-linter"])
+
+    analysis_results = tmp_path / "analysis_results.json"
+    assert analysis_results.exists()
+
+    with open(analysis_results) as file:
+        result = json.load(file)
+
+    assert result == {
+        "issues": [],  # TODO
```
Comment (srijan-deepsource) [inline tests/community_analyzer_test.py]: There's a TODO comment here.

Also, did our Python Analyzer stop catching these?

**Comment by srijan-deepsource** [comment]
Also, set up test coverage reporting?

**Comment by tushar-deepsource** [comment]
> Also, set up test coverage reporting?

Don't have the perms to set secrets on this repo.

> There's a TODO comment here.

Can you give an actual kube linter artifact? I'll write a test against that.

> did our Python Analyzer stop catching these?

I think it only catches it if it's `#TODO: ...`

File: sarif-parser/tests/cli_test.py (hunk 1 above)
Comment (tushar-deepsource) [reply sarif-parser/tests/cli_test.py:114]: Yes, that is made up. `sarif-parser` tests don't deal with real world mappings. Tests regarding that will be in the top level `tests` folder.

**Comment by srijan-deepsource** [comment]
> Don't have the perms to set secrets on this repo.

Added `DEEPSOURCE_DSN` as a secret. Go ahead and make changes in the workflow to report it.

> I think it only catches it if it's #TODO: ...

Can you verify and create issue? It should catch blanket TODOs as well.

**Comment by tushar-deepsource** [comment]
It does get raised on `# TODO` as well. Not sure why it didn't happen in this case.

```python
def foo():
    pass # TODO
```
```console
$ python -m marvin_py -sl /tmp/code/**/*.py            
{"path": "/tmp/code/a.py", "line": 1, "column": 0, "message-id": "PYL-C0104", "message": "Disallowed name \"foo\"", "context": null}
{"path": "/tmp/code/a.py", "line": 2, "column": 10, "message-id": "PYL-W0511", "message": "TODO", "context": null}
```

File: .github/workflows/tox.yml [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+          python-version: 3.11
+          architecture: x64
+
+      - name: Install dependencies
+        run: pip install tox
+
+      - name: Run tests and type checking
+        run: tox
+
+      - name: Report test coverage to DeepSource
```
Comment (srijan-deepsource) [inline .github/workflows/tox.yml:31]: use test cov action

Comment (tushar-deepsource) [reply .github/workflows/tox.yml:31]: Done

**Review by srijan-deepsource — APPROVED** [review]

**Pull request merged**